### PR TITLE
Add Complete interpreter test coverage for ClusterPolicy

### LINKED ISSUE #6952: [Umbrella] Enhance third-party resource interpreter tests with output verification
/kind cleanup

We wish to enhance the testing framework for third-party resource interpreter customizations by adding support for verifying Lua script execution results against expected outputs.

In addition, the current test coverage for Lua script content is insufficient. Each operation of each resource has only one test case, which is not enough to fully cover the logic. Therefore, we also need to supplement the test cases for individual operations of individual resources.

Below, let's list all the subtasks that need to be completed for the current work item.

- [x] enhance resource interpreter test framework @zhzhuang-zju #6938 
- [ ] Complete test cases to cover the Lua script logic
  - [ ] apps.kruise.io
    - [ ] v1alpha1
      - [ ] AdvancedCronJob @XiShanYongYe-Chang #6991 
      - [ ] BroadcastJob
      - [ ] CloneSet
      - [ ] DaemonSet
      - [ ] SidecarSet
      - [ ] UnitedDeployment
    - [ ] v1beta1 - StatefulSet
  - [ ] argoproj.io - v1alpha1 - Workflow
  - [ ] flink.apache.org - v1beta1 - FlinkDeployment
  - [ ] helm.toolkit.fluxcd.io - v2beta1 - HelmRelease
  - [ ] kubeflow.org - v1 - TFJob
  - [ ] kustomize.toolkit.fluxcd.io - v1 - Kustomization
  - [ ] kyverno.io - v1
    - [ ] ClusterPolicy
    - [ ] Policy
  - [ ] source.toolkit.fluxcd.io
    - [ ] v1 - GitRepository
    - [ ] v1beta2
      - [ ] Bucket
      - [ ] HelmChart
      - [ ] HelmRepository
      - [ ] OCIRepository

If anyone is willing to get involved and complete some of these subtasks, we would be very grateful and welcome your participation. 

You can directly tell me in the comments which subtasks you want to complete, and I will assign them to you.

### PR DESCRIPTION
**Pull request by Abhay349**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
This PR adds comprehensive test coverage for the ClusterPolicy resource interpreter customization.

Specifically, it:
	•	Adds unit tests for AggregateStatus, InterpretStatus, and InterpretHealth
	•	Aligns test expectations with the actual aggregation behavior (e.g. preserving spec while aggregating status)
	•	Covers multi-member aggregation, mixed condition reasons, and edge cases such as nil or missing fields

These tests improve confidence in Kyverno ClusterPolicy status handling across clusters and prevent future regressions.

**Which issue(s) this PR fixes**:
Fixes #6952

**Special notes for your reviewer**:
	•	Tests are written to strictly follow the existing Lua aggregation logic
	•	spec is intentionally preserved in aggregated results, as AggregateStatus only mutates status
	•	All cases are verified using:
                 `go test ./pkg/resourceinterpreter/default/thirdparty`

**Does this PR introduce a user-facing change?**:
None
  
Thanks to @FAUST-BENCHOU for review and feedbacks.

### CODE REVIEW

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/interpretstatus-test.yaml [hunk 1]
```diff
@@ -41,3 +44,65 @@ observedObj:
...
+      - lastTransitionTime: "2023-05-07T12:28:58Z"
+        message: ""
+        reason: Succeeded
+        status: "True"
+        type: Ready
+    rulecount:
+      generate: 0
+      mutate: 0
+      validate: 1
+      verifyimages: 0
```
[the remark this answers is not in the document]
Comment (Ferryx349) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/interpretstatus-test.yaml:59]: autogen is omitted because it’s an empty object. The interpreter doesn’t preserve empty maps in the output, which matches existing interpreter test behavior.

**Comment by XiShanYongYe-Chang** [comment]
Thanks a lot @Abhay349 
/cc @FAUST-BENCHOU

**Comment by FAUST-BENCHOU** [comment]
/assign
may takes some time :)

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/interpretstatus-test.yaml [hunk 2]
```diff
@@ -41,3 +44,65 @@ observedObj:
...
+---
+name: "ClusterPolicy with nil status"
+description: "ReflectStatus should return empty status when status is nil"
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/interpretstatus-test.yaml]: nit: The description may be a little confusing since above is `Test interpreting status for ClusterPolicy` (and so is the file name).I undersatnd may because of the `ReflectStatus` in lua.Anyway just unify them so that others can make them clear.

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/interprethealth-test.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+name: "ClusterPolicy unhealthy when conditions are empty"
+description: "Health should be false when status.conditions is empty"
+observedObj:
+  apiVersion: kyverno.io/v1
+  kind: ClusterPolicy
+  status:
+    conditions: []
+operation: InterpretHealth
+output:
+  healthy: false
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/interprethealth-test.yaml:82]: We can also add another test case that there is not even status fields,and it will return false directly
https://github.com/karmada-io/karmada/blob/b97876796aec8241929f54a0072656b7e09cfa6f/pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/customizations.yaml#L16

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/aggregatestatus-test.yaml [hunk 4]
```diff
@@ -58,3 +60,95 @@ statusItems:
...
+          status: "True"
+          reason: Failed
+          message: "member2=error"
+      rulecount:
+        generate: 0
+        mutate: 0
+        validate: 0
+        verifyimages: 0
+---
+name: "ClusterPolicy with nil statusItems"
```
Comment (FAUST-BENCHOU) [inline pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/aggregatestatus-test.yaml:141]: I'm curious about whether `statusItems :nil` is the same with `statusItems: []`.Seems when statusItems is empty array will still enter the `for i=1,#statusItems loop`.So the fields like `rulecount` may also be set to 0.Anyway thats just a nit

Comment (Ferryx349) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/aggregatestatus-test.yaml:141]: In this Lua implementation, `statusItems: nil` and `statusItems: []` behave slightly differently. When statusItems == nil, the function returns early and doesn’t initialize status or rulecount at all.
With an empty array, it passes the nil check, initializes status/rulecount, and then skips the loop since the array is empty, so the fields end up zeroed.

This test is intentionally covering the early-return (nil) case to make that behavior explicit. Happy to rename it for clarity or add a separate test for the empty-array case if you think that would be clearer...

Thnaks :)

**Comment by Ferryx349** [comment]
@FAUST-BENCHOU thanks for review and feedback. I have fixed the recommended changes, let me know if there are issues..
Happy to iterate on further feedback..

Thanks again!

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/aggregatestatus-test.yaml (hunk 4 above)
Comment (FAUST-BENCHOU) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/aggregatestatus-test.yaml:141]: not a big deal.just keep it

**Comment by FAUST-BENCHOU** [comment]
lgtm.Thanks for your work!
@XiShanYongYe-Chang @Abhay349

File: pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/aggregatestatus-test.yaml (hunk 4 above)
Comment (Ferryx349) [reply pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/kyverno.io/v1/ClusterPolicy/testdata/aggregatestatus-test.yaml:141]: Got it.

**Comment by XiShanYongYe-Chang** [comment]
Thanks a lot @FAUST-BENCHOU
